skrub/_adaptive_squashing.py

```diff
@@ -0,0 +1,207 @@
+            raise ValueError(
+                "upper_quantile_alpha needs to be a number, but got"
+                f" type(upper_quantile_alpha)={type(self.upper_quantile_alpha)}"
+            )
+        if self.max_absolute_value <= 0 or not np.isfinite(self.max_absolute_value):
+            raise ValueError(
+                "max_absolute_value needs to be a positive finite float, but got"
+                f" {self.max_absolute_value}"
+            )
+        if not (0 <= self.lower_quantile_alpha < self.upper_quantile_alpha <= 1):
+            raise ValueError(
+                "need 0 <= lower_quantile_alpha < upper_quantile_alpha <= 1, but got"
+                f" lower_quantile_alpha={self.lower_quantile_alpha} and"
+                f" upper_quantile_alpha={self.upper_quantile_alpha}"
+            )
+
+        self.col_name_ = sbd.name(X)
+
+        if not sbd.is_numeric(X):
+            raise RejectColumn(f"Column {self.col_name_} is not numeric.")
+
+        eps = 1e-30  # todo: expose?
+
+        values = sbd.to_numpy(X).astype(np.float32)
+        finite_values = values[np.isfinite(values)]
+        if len(finite_values) > 0:
+            self.median_ = np.median(finite_values)
+            quantiles = [
+                np.quantile(finite_values, q)
+                for q in [self.lower_quantile_alpha, self.upper_quantile_alpha]
+            ]
+            if quantiles[1] == quantiles[0]:
+                min = np.min(finite_values)
+                max = np.max(finite_values)
+                if min == max:
+                    self.scale_ = 0
+                else:
+                    self.scale_ = 2.0 / (max - min + eps)
+            else:
+                self.scale_ = 1.0 / (quantiles[1] - quantiles[0] + eps)
+        else:
+            self.median_ = 0.0
+            self.scale_ = 1.0
+
+        self._is_fitted = True
+        self.n_components_ = 1
+
+        return self.transform(X)
+
+    def transform(self, X):
+        """Transform a column.
+
+        Parameters
+        ----------
+        X : Pandas or Polars series
+            The column to transform.
+
+        Returns
+        -------
+        result: Pandas or Polars column with shape (len(X),)
+            The scaled and squashed representation of the input.
+        """
+
+        values_np = sbd.to_numpy(X).astype(np.float32, copy=True)
+        isfinite = np.isfinite(values_np)
+        scaled_finite = self.scale_ * (values_np[isfinite] - self.median_)
+        values_np[isfinite] = scaled_finite / np.sqrt(
+            1 + (scaled_finite / self.max_absolute_value) ** 2
+        )
+        isinf = np.isinf(values_np)
+        values_np[isinf] = np.sign(values_np[isinf]) * self.max_absolute_value
+
+        # this will use the column name of the column that was passed in fit_transform,
+        # not the name of X
+        return self._post_process(X, values_np, self.col_name_)
+
+    def _post_process(self, X, result, name):
```

Comment:
if this function is only called in 1 place we can just inline those 2 lines and not define a function for it